### PR TITLE
Add support for DataFrame.isna method

### PR DESCRIPTION
**Pull request by kozlov-alexey**

### CODE REVIEW

File: hpat/hiframes/dataframe_pass.py [hunk 1]
```diff
@@ -1357,6 +1372,31 @@ def _run_call_drop(self, assign, lhs, rhs):
...
+        func_text = "def _isna_impl({}):\n".format(", ".join(data_args))
+        for d in data_args:
+            func_text += "  {} = hpat.hiframes.api.init_series({})\n".format(d + '_S', d)
+            func_text += "  {} = hpat.hiframes.api.get_series_data({}.isna())\n".format(d + '_O', d + '_S')
+        func_text += "  return hpat.hiframes.pd_dataframe_ext.init_dataframe({}, None, {})\n".format(
+            ", ".join(d + '_O' for d in data_args),
+            ", ".join("'{}'".format(c) for c in df_typ.columns))
+
+        loc_vars = {}
+        exec(func_text, {}, loc_vars)
```
Comment (shssf) [inline hpat/hiframes/dataframe_pass.py]: Just to clarify the function generated here:
```
def _isna_impl(data0, data1, data2, data3):
  data0_S = hpat.hiframes.api.init_series(data0)
  data0_O = hpat.hiframes.api.get_series_data(data0_S.isna())
  data1_S = hpat.hiframes.api.init_series(data1)
  data1_O = hpat.hiframes.api.get_series_data(data1_S.isna())
  data2_S = hpat.hiframes.api.init_series(data2)
  data2_O = hpat.hiframes.api.get_series_data(data2_S.isna())
  data3_S = hpat.hiframes.api.init_series(data3)
  data3_O = hpat.hiframes.api.get_series_data(data3_S.isna())
  return hpat.hiframes.pd_dataframe_ext.init_dataframe(data0_O, data1_O, data2_O, data3_O, None, 'A', 'B', 'C', 'D')
```
It is not a request for changes but, usually, I put an example of generated data in such cases to comments.

File: hpat/tests/test_dataframe.py [hunk 2]
```diff
@@ -606,6 +606,18 @@ def test_impl():
...
             hpat.hiframes.sort.MIN_SAMPLES = save_min_samples
 
+    def test_df_isna1(self):
+        def test_impl(df):
+            return df.isna()
+        hpat_func = hpat.jit(test_impl)
+
+        # TODO: add column with datetime values when test_series_datetime_isna1 is fixed
+        df = pd.DataFrame({'A': [1.0, 2.0, np.nan, 1.0],
+                           'B': [np.nan, 5, 6, np.nan],
```
Comment (shssf) [inline hpat/tests/test_dataframe.py]: it might be good to add `np.inf` here also

File: hpat/hiframes/dataframe_pass.py [hunk 3]
```diff
@@ -1357,6 +1372,31 @@ def _run_call_drop(self, assign, lhs, rhs):
...
+        n_cols = len(df_typ.columns)
+        data_args = tuple('data{}'.format(i) for i in range(n_cols))
+
+        func_text = "def _isna_impl({}):\n".format(", ".join(data_args))
+        for d in data_args:
+            func_text += "  {} = hpat.hiframes.api.init_series({})\n".format(d + '_S', d)
+            func_text += "  {} = hpat.hiframes.api.get_series_data({}.isna())\n".format(d + '_O', d + '_S')
+        func_text += "  return hpat.hiframes.pd_dataframe_ext.init_dataframe({}, None, {})\n".format(
+            ", ".join(d + '_O' for d in data_args),
+            ", ".join("'{}'".format(c) for c in df_typ.columns))
```
Comment (densmirn) [inline hpat/hiframes/dataframe_pass.py]: I would propose to create list of lines. Then join the list using `\n`. E.g.:
`func_lines = ['first_line']`
`func_lines.append('second_line')`
`func_lines.append('third_line')`
`func_text = '\n'.join(func_lines) # func_text = os.linesep.join(func_lines)` ([os.linesep](https://docs.python.org/3/library/os.html#os.linesep))

When concatenating strings in Python new string is created each time. E.g. here 5 strings will be created (`he`, `ll`, `hell`, `o`, `hello`):
`a = 'he'`
`a += 'll'`
`a += 'o'`

File: hpat/hiframes/dataframe_pass.py (hunk 1 above)
Comment (kozlov-alexey) [reply hpat/hiframes/dataframe_pass.py]: I looked how a multi-line comment just before exec looks, but it actually makes code look over complicated, so I decided that's it's better without it. Using simple "print(func_text)" is sufficient most of the times.

File: hpat/tests/test_dataframe.py (hunk 2 above)
Comment (kozlov-alexey) [reply hpat/tests/test_dataframe.py]: Added

File: hpat/hiframes/dataframe_pass.py (hunk 3 above)
Comment (kozlov-alexey) [reply hpat/hiframes/dataframe_pass.py]: Good point, rewritten it in the last commit.

**Pull request merged**